presto/presto_test.go

```diff
@@ -324,6 +341,28 @@ func TestUnsupportedExec(t *testing.T) {
 	}
 }
 
+func TestJWTAuthHeader(t *testing.T) {
+	// this test ensures that the JWT token is passed as a Bearer token within the Authorization header
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		// validate the Authorization header is JWT token
+		if r.Header.Get("Authorization") != "Bearer test_token" {
+			t.Fatal("unexpected Authorization header, want JWT token")
```

Comment:
I feel it is better to return an error by writing a status code here rather than failing the test inside the server.
```
if r.Header.Get("Authorization") != "Bearer test_token" {
	w.WriteHeader(http.StatusInternalServerError)
} else {
	w.WriteHeader(http.StatusOK)
}
```

Reply:
thanks for the suggestion, I think both implementations are still a little awkward so I'm curious if there is a better approach overall.

my previous implementation makes it clear the bearer token is not what is expected but as you said, requires the test to fail in the http handler. I agree the suggested implementation is better as the http handler failing is quite odd and have gone ahead and implemented with a slight modification.

the main downside I see is that a 500 internal error could be more difficult to debug / doesn't make sense if `ts` is acting as a presto server. one compromise I thought of was to return `http.StatusUnauthorized` because it could represent the proxy in front of presto returning an http error code. please let me know your thoughts if you think there is a better solution!

Reply:
What you came up with looks good to me!